src/tests/Search.test.tsx

```diff
@@ -0,0 +1,37 @@
+import { Search } from '../components/Search.tsx'
+import { render, screen } from '@testing-library/react'
+import { userEvent } from '@testing-library/user-event'
+
+describe('Search', () => {
+    const user = userEvent.setup()
+    const initialValue = 'initial value'
+    const newValue = 'new value'
+    const onSearch = jest.fn()
```

Comment:
You are instantiating a mock function outside of a test and the reusing it in different tests. It may lead to bugs, you should either write a setup function or clear the mocks after each test.